test/test_util.py

```diff
@@ -191,141 +181,133 @@ def test_parse_url_normalization(self):
         # Empty Port
         'http://google.com:': Url('http', host='google.com'),
         'http://google.com:/': Url('http', host='google.com', path='/'),
+    }
 
-        }
+    @pytest.mark.parametrize(
+        'url, expected_Url',
+        chain(parse_url_host_map, non_round_tripping_parse_url_host_map.items())
+    )
+    def test_parse_url(self, url, expected_Url):
+        returned_Url = parse_url(url)
+        assert returned_Url == expected_Url
 
-    def test_parse_url(self):
-        for url, expected_Url in chain(self.parse_url_host_map,
-                                       self.non_round_tripping_parse_url_host_map.items()):
-            returned_Url = parse_url(url)
-            self.assertEqual(returned_Url, expected_Url)
-
-    def test_unparse_url(self):
-        for url, expected_Url in self.parse_url_host_map:
-            self.assertEqual(url, expected_Url.url)
+    @pytest.mark.parametrize('url, expected_Url', parse_url_host_map)
+    def test_unparse_url(self, url, expected_Url):
```

Comment:
The casing isn’t entirely without merit – it’s looking for instances of the `Url` class.